src/query/api/v1/handler/prometheus/remote/attribution.go

```diff
@@ -0,0 +1,84 @@
+package remote
+
+import (
+	"errors"
+	"math/rand"
+	"strings"
+	"sync"
+	"sync/atomic"
+
+	"github.com/m3db/m3/src/query/generated/proto/prompb"
+	"github.com/m3db/m3/src/x/instrument"
+	"github.com/uber-go/tally"
+)
+
+/**
+ * This class attributes metrics with matched configured labels for visibility purpose.
+ * For example, today m3 only emits totally write QPS, with this attribution, we could plot
+ * QPS breakdown by user defined label like namespace in Kubernetes or any label with finite
+ * values (configured via capacity) and business meanings.
+ */
+
+var errNoOption = errors.New("no option configured for promAttributionMetrics")
+
+type promAttributionMetrics struct {
+	opts               *instrument.AttributionConfiguration
+	baseScope          tally.Scope
+	attributedCounters sync.Map
+	counterSize        int32
+	skipSamples        tally.Counter
+	missSamples        tally.Counter
+}
+
+func (pam *promAttributionMetrics) attribute(ts prompb.TimeSeries) {
+	if rand.Float64() >= pam.opts.SamplingRate {
+		return
+	}
+	labelValues := make([]string, len(pam.opts.Labels))
+	found := 0
+	sample_count := int64(len(ts.Samples))
+	for _, l := range ts.Labels {
+		labelName := string(l.Name)
+		for i, label := range pam.opts.Labels {
+			if labelName == label {
+				found++
+				labelValues[i] = string(l.Value)
+			}
+		}
+	}
+	if found < len(pam.opts.Labels) {
```

Comment:
wondering does it have to be match all labels? e.g: for some labels that may only exist in DP or CP.

Reply:
That's the idea of match, you specify 1 label to match or multiple labels, for example, I have metrics with `namespace="abc"`, and another one is `proxy="k2p", workspaceId="12342"`, in this case, we can configure two attributions:
```
attributions:
- name: cp
  capacity: 100
  samplingRate: 0.1
  labels: ["namespace"]
- name: dp
  capacity: 1000
  samplingRate: 0.1
  labels: ["proxy", "workspaceId"]
```

Reply:
ideally we could add filtering logic as well similar to samplingRate